WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -475,15 +475,15 @@ public async Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(Connect
 				}
 			}
 
-			var amountZeroTask = round.AmountCredentialIssuer.PrepareResponse(request.ZeroAmountCredentialRequests, cancellationToken);
-			var vsizeZeroTask = round.VsizeCredentialIssuer.PrepareResponse(request.ZeroVsizeCredentialRequests, cancellationToken);
-			Task<ICommitableCredentialsResponse>? amountRealTask = null;
-			Task<ICommitableCredentialsResponse>? vsizeRealTask = null;
+			var amountZeroTask = round.AmountCredentialIssuer.HandleRequest(request.ZeroAmountCredentialRequests, cancellationToken);
```

Comment:
When we added this Prepare/Commit strategy the problem we solved was to avoid having something like a CredentialIssuer Undo if an exception happens after we processed the request. How are we protected against that now?

Reply:
See #6086. When serial numbers are formally valid, the serial numbers are cryptographically unguessable, and only formally valid serial numbers are added to the nullifier set. Because they are unguessable the only way that a serial number will be presented twice is if a client is constructing malformed requests, because otherwise the idempotent handler would catch it. The nullifier set is prevented from being filled by only permanently storing valid serial numbers (which correspond to previously issued credentials, which are rate limited by UTXO confirmation).

Because there is no sense in making affordances for malicious clients, Postel's maxim can't really be applied in a logically consistent way in an adverserial setting, any partially valid spend attempt can be seen as actively malicious so there is no point in trying to accomodate a subsequent request that is valid, same logic as #6161, #6162

Reply:
In fact, thinking about it a bit more, when `HandleRequest` errors during an `Alice` associated request, the input should be banned.

Reply:
Isn't it dangerous to assume every error that can occur is a malicious client action? 
https://github.com/zkSNACKs/WalletWasabi/pull/6181/files#r689454280